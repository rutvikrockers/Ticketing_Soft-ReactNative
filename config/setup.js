// create our store for testing redux components
import configureMockStore from 'redux-mock-store';
const thunk = require('redux-thunk');

// Setup adapter to work with enzyme 3.2.0
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

export const mockStore = configureMockStore([thunk.default]);

Enzyme.configure({ adapter: new Adapter() });