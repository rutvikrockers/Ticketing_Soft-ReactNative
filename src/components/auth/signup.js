import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View
} from 'react-native';
import {Redirect, Link} from 'react-router-native';
import {newUser} from '../../actions';
import {connect} from 'react-redux';
import Button from '../shared/button';
import Input from '../shared/input';

import styles from '../../styles/main.js';

class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {phone: "", email: ""}
  }
  createUser() {
    this.props.dispatch(newUser({
      email: this.state.email,
      password: this.state.phone
    }));
  }
  render() {
    return (
      <View style={styles.container}>
      <Input
        value={this.state.email}
        placeholder="email@example.com"
        onChange={(email) => this.setState({email})}
        secure={false}
      />
      <Input
        placeholder="Phone Number"
        onChange={(phone) => this.setState({phone})}
        value={this.state.phone}
      />
      <View style={{marginTop: 20}} />
        <Button
          style={styles.btnContainer}
          onPress={() => { this.createUser() }}
          value="SIGN UP"
        />
        <Link to="/">
            <Text>Sign In</Text>
        </Link>
      </View>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    user: state.user,
    error: state.error,
  }
}

export default connect(mapStateToProps)(Signup);
