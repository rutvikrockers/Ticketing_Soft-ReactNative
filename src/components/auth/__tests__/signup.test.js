import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../config/setup';
import toJson from 'enzyme-to-json';
import { default as Signup } from '../signup';

const render = (props) => shallow(
    <Provider store={mockStore({})}>
        <Signup {...props} />
    </Provider>
);

describe('Signup', () => {
    let component;
    beforeEach(() => {
        component = render({});
    });

    it('matches its snapshot', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('renders', () => {
        expect(component.find(Signup).length).toBe(1);
    });
});