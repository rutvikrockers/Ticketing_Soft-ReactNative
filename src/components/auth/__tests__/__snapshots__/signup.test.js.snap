// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup matches its snapshot 1`] = `<Connect(Signup) />`;
